### ISSUE TITLE
circular imports, Lazytype, Generics

### ISSUE THREAD

**Issue by moshir** [strawberry-graphql/strawberry#1158]
## Issue 
Can't seem to get simple model working involving 

1. a Generic type (Collection)

```python
# coll.py 
from typing import Generic,Optional,TypeVar,List
import strawberry

T = TypeVar("T")
@strawberry.type
class Collection(Generic[T]):
    items:List[T]
```
2. A simple strawberry type with one Collection field referencing a type (Bar) in another file using string annotation to avoid circular imports

```python
#foo.py
import strawberry
from coll import Collection

@strawberry.type
class Foo:
    id:strawberry.ID
    bars:Collection["Bar"]
````
3. a simple strawberry type 

```python
#bar.py
import strawberry

@strawberry.type
class Bar:
    id:strawberry.ID

```

When defining the schema as follows:
```python
#schema.py
from __future__ import annotations
from typing import Optional
import strawberry
from foo import Foo
from bar  import Bar

def get_foo()->Foo:
    pass

@strawberry.type
class Query:
    getFoo:Optional[Foo]= strawberry.field(get_foo)


s= strawberry.schema.Schema(types=[Foo,Bar],query=Query)
print(s)
```

it breaks with  :
 ```
 raise TypeError(f"{self.name} fields cannot be resolved. {error}")
TypeError: Foo fields cannot be resolved. Unexpected type 'None'
````
### using LazyType : 
```python
#foo.py 
import strawberry
from coll import Collection
from strawberry.lazy_type import LazyType

@strawberry.type
class Foo:
    id:strawberry.ID
    bars:Collection[LazyType["Bar","bar"]]


```
but then got : 
```
TypeError: Foo fields cannot be resolved. 'LazyType' object has no attribute '__name__'
```

### removing __future__ annotations 

same results 
```
TypeError: Foo fields cannot be resolved. 'LazyType' object has no attribute '__name__'
```

Any clue ?

**Comment by patrick91** [comment]
It seems we might have a bug, I tested this code locally and it works with this change:

```patch
diff --git a/strawberry/annotation.py b/strawberry/annotation.py
index a45bc04..474945b 100644
--- a/strawberry/annotation.py
+++ b/strawberry/annotation.py
@@ -63,7 +63,7 @@ class StrawberryAnnotation:
         if self._is_async_generator(evaled_type):
             evaled_type = self._strip_async_generator(evaled_type)
         if self._is_lazy_type(evaled_type):
-            return evaled_type
+            return evaled_type.resolve_type()

         if self._is_generic(evaled_type):
             if any(is_type_var(type_) for type_ in evaled_type.__args__):
```

@BryceBeagle does this change make sense to you? I can send a PR if so :D

**Comment by BryceBeagle** [comment]
Hmm, that specific change breaks `LazyType` in cyclical import scenarios (try running the `test_cyclic_import` test with that patch applied). I'll look into a further situation.

**Comment by BryceBeagle** [comment]
@moshir, this should be fixed in #1235. Can you make sure that it fixes your issues?
